### ISSUE TITLE
Authentication / authorization and granular permissions and user roles

### ISSUE THREAD

**Issue by njhillson** [Global-Biofoundries-Alliance/DNA-scanner#4]
It will be important to set up authentication and authorization with granular permissions controls and user roles.

We will likely want to keep subsets of the data private, private to specific groups, with some people only able to read, others able to read and write, etc.

Will need to have different types of user roles, like admin, or curator, or viewer, etc.

These permissions should be enforced for all channels of accessing and modifying data, e.g. API end points.

**Comment by Zulko** [comment]
To make initial development faster, the project could start as a single webpage without user accounts, a bit like skyscanner. Users would drop sequences and get suggestions.

**Comment by neilswainston** [comment]
Guess we'll go through these tickets and prioritise them. I agree with Valentin that this is likely to be an add-on for later.

**Comment by njhillson** [comment]
I do not think that people will be willing to upload data without authorization/authentication. It also opens bad security holes if you can upload things without some type of account vetting and attribution ability.

Further, I do not think that vendor APIs will allow people to use them without some type of token or other kind of validation. Perhaps we see as we go, but I think that this is a higher priority than may otherwise be acknowledged.

**Comment by neilswainston** [comment]
OK. Fair point. Twist certainly require authentication to access ther API.
